### PR TITLE
[Feature] Make StarRocks support FQDN

### LINKED ISSUE #5527: Make StarRocks support FQDN
## Enhancement
FQDN means Fully Qualified Domain Name, StarRocks only support IP now, we need to make StarRocks support FQDN

### PR DESCRIPTION
**Pull request by waittttting**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #5527 #401 

## Problem Summary(Required):

StarRocks only supported IP before. Some users' network environments need to support FQDN.
You can refer to the following documentation to use FQDN in your StarRocks cluster
First, you need to configure all the FQDN information of the cluster in each machine's /etc/hosts file.
If you want to use FQDN to create a new cluster,  you should do nothing, this feature will make cluster use fqdn automatically 
And if you want to upgrade an old cluster, please follow the next steps.

### **Be change ip to fqdn**
1. BE Reboot directly after upgrading to the version that contains the feature
2. FE Execute SQL
alter system modify backend host "192.1.1.0" to "testFqdn"

### **Fe upgrade ip to fqdn**
- FE cluster upgrade with less than three followers is not supported
- Change follower node first, then master node

**How to follower's ip to fqdn?**
1. stop the node you want to change the IP to FQDN, wait for the alive state to be false
2. Execute the following SQL
alter system modify frontend host "127.1.1.0" to "sandbox16"
3. After the FE is upgraded to the version that includes the feature
4. restart the FE with `sh bin/start_fe.sh --host_type=fqdn` --daemon and wait for the FE alive state to be true

**How to master's ip to fqdn?**

When all the follower besides of the master in the cluster are upgraded, shut down the master and wait for the new master to be elected, and then restart  the original master, wait it to be a follower and alive. 
Then the new follower could be processed according to the above to upgrade ip to fqdn

## Other:

this feature can help you start fe by using specified host type，just like `sh bin/start_fe.sh --host_type=fqdn` `sh bin/start_fe.sh --host_type=ip`

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 228 / 281 (81.14%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/alter/Alter.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/StarRocksFE.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/qe/DdlExecutor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/alter/SystemHandler.java|4|7|57.14%|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeHandler.java|4|7|57.14%|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|3|5|60.00%|
|:large_blue_circle:|com/starrocks/server/NodeMgr.java|33|44|75.00%|
|:large_blue_circle:|com/starrocks/service/FrontendOptions.java|60|76|78.95%|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|4|5|80.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AlterSystemStmtAnalyzer.java|11|13|84.62%|
|:large_blue_circle:|com/starrocks/system/SystemInfoService.java|45|52|86.54%|
|:large_blue_circle:|com/starrocks/common/proc/FrontendsProcNode.java|7|8|87.50%|
|:large_blue_circle:|com/starrocks/analysis/ModifyFrontendAddressClause.java|8|9|88.89%|
|:large_blue_circle:|com/starrocks/analysis/ModifyBackendAddressClause.java|8|9|88.89%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/common/proc/BackendsProcDir.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/persist/Storage.java|11|11|100.00%|
|:large_blue_circle:|com/starrocks/system/Backend.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/catalog/BrokerMgr.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/common/util/NetUtils.java|11|11|100.00%|
|:large_blue_circle:|com/starrocks/system/Frontend.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/ha/BDBHA.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/alter/MaterializedViewHandler.java|1|1|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java [hunk 1]
```diff
@@ -374,8 +374,16 @@ public Backend getBackendWithBePort(String host, int bePort) {
                 LOG.info("faild to get address by fqdn {}", e.getMessage());
```
Comment (Astralidea) [inline fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java:374]: better to add UT for getBackendWithBePort and mock idToBackend to cover ip or FQDN set or not be select logical.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 247 / 306 (80.72%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/alter/Alter.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/DdlExecutor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/alter/SystemHandler.java|4|7|57.14%|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeHandler.java|4|7|57.14%|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|3|5|60.00%|
|:large_blue_circle:|com/starrocks/StarRocksFE.java|15|24|62.50%|
|:large_blue_circle:|com/starrocks/server/NodeMgr.java|33|44|75.00%|
|:large_blue_circle:|com/starrocks/service/FrontendOptions.java|63|79|79.75%|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|4|5|80.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AlterSystemStmtAnalyzer.java|11|13|84.62%|
|:large_blue_circle:|com/starrocks/system/SystemInfoService.java|45|52|86.54%|
|:large_blue_circle:|com/starrocks/common/proc/FrontendsProcNode.java|7|8|87.50%|
|:large_blue_circle:|com/starrocks/analysis/ModifyFrontendAddressClause.java|8|9|88.89%|
|:large_blue_circle:|com/starrocks/analysis/ModifyBackendAddressClause.java|8|9|88.89%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/common/proc/BackendsProcDir.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/persist/Storage.java|11|11|100.00%|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/system/Backend.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/catalog/BrokerMgr.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/common/util/NetUtils.java|11|11|100.00%|
|:large_blue_circle:|com/starrocks/system/Frontend.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/ha/BDBHA.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/alter/MaterializedViewHandler.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 247 / 306 (80.72%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/alter/Alter.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/DdlExecutor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/alter/SystemHandler.java|4|7|57.14%|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeHandler.java|4|7|57.14%|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|3|5|60.00%|
|:large_blue_circle:|com/starrocks/StarRocksFE.java|15|24|62.50%|
|:large_blue_circle:|com/starrocks/server/NodeMgr.java|33|44|75.00%|
|:large_blue_circle:|com/starrocks/service/FrontendOptions.java|63|79|79.75%|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|4|5|80.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AlterSystemStmtAnalyzer.java|11|13|84.62%|
|:large_blue_circle:|com/starrocks/system/SystemInfoService.java|45|52|86.54%|
|:large_blue_circle:|com/starrocks/common/proc/FrontendsProcNode.java|7|8|87.50%|
|:large_blue_circle:|com/starrocks/analysis/ModifyFrontendAddressClause.java|8|9|88.89%|
|:large_blue_circle:|com/starrocks/analysis/ModifyBackendAddressClause.java|8|9|88.89%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/common/proc/BackendsProcDir.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/persist/Storage.java|11|11|100.00%|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/system/Backend.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/catalog/BrokerMgr.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/common/util/NetUtils.java|11|11|100.00%|
|:large_blue_circle:|com/starrocks/system/Frontend.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/ha/BDBHA.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/alter/MaterializedViewHandler.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 247 / 306 (80.72%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/alter/Alter.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/DdlExecutor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/alter/SystemHandler.java|4|7|57.14%|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeHandler.java|4|7|57.14%|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|3|5|60.00%|
|:large_blue_circle:|com/starrocks/StarRocksFE.java|15|24|62.50%|
|:large_blue_circle:|com/starrocks/server/NodeMgr.java|33|44|75.00%|
|:large_blue_circle:|com/starrocks/service/FrontendOptions.java|63|79|79.75%|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|4|5|80.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AlterSystemStmtAnalyzer.java|11|13|84.62%|
|:large_blue_circle:|com/starrocks/system/SystemInfoService.java|45|52|86.54%|
|:large_blue_circle:|com/starrocks/common/proc/FrontendsProcNode.java|7|8|87.50%|
|:large_blue_circle:|com/starrocks/analysis/ModifyFrontendAddressClause.java|8|9|88.89%|
|:large_blue_circle:|com/starrocks/analysis/ModifyBackendAddressClause.java|8|9|88.89%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/common/proc/BackendsProcDir.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/persist/Storage.java|11|11|100.00%|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/system/Backend.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/catalog/BrokerMgr.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/common/util/NetUtils.java|11|11|100.00%|
|:large_blue_circle:|com/starrocks/system/Frontend.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/ha/BDBHA.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/alter/MaterializedViewHandler.java|1|1|100.00%|

**Review by Astralidea — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Pull request merged**